src/dns_internal.cc

```diff
@@ -833,7 +833,7 @@ idnsDoSendQueryVC(nsvc *vc)
     vc->busy = 1;
 
     // Comm needs seconds but idnsCheckQueue() will check the exact timeout
-    const int timeout = (Config.Timeout.idns_query % 1000 ?
+    const time_t timeout = (Config.Timeout.idns_query % 1000 ?
```

Comment:
The actual type here is currently time_msec_t rather than time_t. Those two types are rather different (e.g., only one of them is _unsigned_ in most environments). If the compiler is happy, I suggest removing this type conversion from _this_ line because this code is not written to handle that conversion correctly:

```suggestion
    const auto timeout = (Config.Timeout.idns_query % 1000 ?
```

The above will also improve code consistency: There are quite a few similar `auto` lines in the existing code. I found only one other line that uses an explicit time_t variable like this PR code does.